demos/tutorials/Deconvolution1D.py

```diff
@@ -0,0 +1,164 @@
+"""
+One-dimensional deconvolution
+=============================
+
+    This tutorial walks through the process of solving a simple deconvolution problem
+    in a Bayesian setting.
+
+"""
+# %%
+# Setup
+# -----
+# We start by importing the necessary modules
+
+import cuqi
+import numpy as np
+import matplotlib.pyplot as plt
+
+# %%
+# Setting up the forward model
+# ----------------------------
+# We start by defining the forward model. In this case, we will use a simple
+# convolution model. The forward model is defined by the following equation:
+#
+# .. math::
+#    \mathbf{y} = \mathbf{A} \mathbf{x}
+#
+# where :math:`\mathbf{y}` is the data, :math:`\mathbf{A}` is the convolution (forward model)
+# operator, and :math:`\mathbf{x}` is the solution.
+#
+# The easiest way to define the forward model is to use the testproblem module.
+# This module contains a number of pre-defined test problems that contain the
+# forward model and synthetic data. In this case, we will use the
+# :class:`cuqi.testproblem.Deconvolution1D` test problem. We extract the forward model
+# and synthetic data from the test problem by calling the :func:`get_components` static
+# method.
+
+# Forward model and data
+A, y_data, info = cuqi.testproblem.Deconvolution1D.get_components(use_legacy=False)
+
+# %%
+# There are many parameters that can be set when creating the test problem. In this case
+# we will use the default parameters. The :func:`get_components` method returns the
```

Comment:
Bit surprising to see use_legacy set to false as the very first thing - I would expect this to be the default, so not necessary to specify?  I would probably also expect a Deconvolution1D tutorial to demonstrate a few more features than just the default, either right away, or as a later example of the tutorial?

Reply:
Ah, forgot to remove that. Its default false now :)